applications/nrf_desktop/src/events/hid_out_event.c

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdio.h>
+
+#include "hid_out_event.h"
+
+
+static int log_hid_out_report_event(const struct event_header *eh, char *buf, size_t buf_len)
```

Comment:
why can't we use hid_report_event?

Reply:
I prefer to use separate event, because it represents data from host and therefore it's handled by the application in completely different way. If we would like to use hid_report_event instead, we would need to check if it's related to HID input/output report and ignore it if it's relevant for us. The check would be needed for every handler. I think that currently used approach is simpler.

Reply:
> I think that currently used approach is simpler.

It's not simpler. It's more complicated. The only advantage is that you may gain slight performance improvement - depending on how much checking report type cost. I know it will cost something but I don't think it will be enough to notice anywhere.

> it's handled by the application in completely different way
It's HID report. Application modules could possibly do whatever it needs with it. You could have a module that uses HID report (connected with input) for internal purposes (e.g. changing brightness depending on how fast mouse moves). Output report carries exactly the same information as input report.

> we would need to check if it's related to HID input/output report and ignore it if it's relevant for us
Sure. This would be the cost we pay for having a single report type and simplified maintenance.